src/components/scrollable-list/scrollable-list.component.js

```diff
@@ -0,0 +1,156 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import tagComponent from '../../utils/helpers/tags';
+import Events from '../../utils/helpers/events';
+import ScrollableItemWrapper from './scrollable-item-wrapper.component';
+import ScrollableListContext from './scrollable-list.context';
+import ScrollableListContainer from './scrollable-list.style';
+
+
+class ScrollableList extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    onSelect: PropTypes.func,
+    onLazyLoad: PropTypes.func,
+    keyNavigation: PropTypes.bool,
+    maxHeight: PropTypes.string
+  };
+
+  state = {
+    selectedItem: null
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.keyNavigation && !this.props.keyNavigation) {
+      document.removeEventListener('keydown', this.handleKeyDown);
+    } else if (!prevProps.keyNavigation && this.props.keyNavigation) {
+      document.addEventListener('keydown', this.handleKeyDown);
+    }
+  }
+
+  scrollBox = React.createRef();
+
+  componentDidMount() {
+    if (this.props.keyNavigation) {
+      document.addEventListener('keydown', this.handleKeyDown);
+    }
+    const initialSelectedItem = this.nextSelectable('down', -1);
```

Comment:
does this mean the 1st item is automatically highlighted? it might be better to make nothing highlighted by default (eg. `-1`?) - if we do this then you can remove this and just add it to the initial state object on line 20.

Reply:
I presumed that if, for example, the user types an `ab`, it might be more expected that the closest or top match (i.e. `abacus`) would be immediately selected for use?

Reply:
I think thats true as they type, but not on open - see https://carbon.sage.com/components/dropdown-filter

It may be that the select/option components will dictate this. Leave it for now and we will see how it works when we plug it all together 👍